### DISCUSSION TITLE
`trace` throwing in production builds in MobX 6 is a breaking change

### DISCUSSION THREAD

**Discussion by quanganhtran** [mobxjs/mobx#3708] [General] [answered]
In ea0b1d8, `trace` started throwing in production and this does not seem to be mentioned anywhere. Even if `trace` was never meant to be used in production, it ran just fine before so this should be marked as a breaking change.

This commit seemed to be a part of MobX 6 release. At least it did not throw in MobX 5.15.0

**Comment by quanganhtran** [comment]
Also I am not sure why it has to throw an error instead of being a noop

**Comment by kubk** [comment, marked as answer]
Thanks for the report. I've provided a fix. Let's wait until it is merged: https://github.com/mobxjs/mobx/pull/3709
